tensorbay/client/requests.py

```diff
@@ -317,17 +323,73 @@ def multithread_upload(
     Arguments:
         function: The upload function.
         arguments: The arguments of the upload function.
+        callback_function: The callback function.
         jobs: The number of the max workers in multi-thread uploading procession.
         pbar: The :class:`Tqdm` instance for showing the upload process bar.
 
     """
     with ThreadPoolExecutor(jobs) as executor:
+        if callback_function is not None:
+            multi_callback = MultiCallbackMission(
+                function=function, callback_function=callback_function
+            )
+            function = multi_callback.mission  # type: ignore[assignment]
         futures = [executor.submit(function, argument) for argument in arguments]
+
         for future in futures:
             future.add_done_callback(pbar.update_callback)
 
         done, not_done = wait(futures, return_when=FIRST_EXCEPTION)
+
+        if callback_function is not None:
+            callback_arguments = [
+                multi_callback.arguments.get() for _ in range(multi_callback.arguments.qsize())
+            ]
+            if len(callback_arguments) > 0:
+                multi_callback.callback_function(callback_arguments)
+
         for future in not_done:
             future.cancel()
         for future in done:
             future.result()
+
+
+class MultiCallbackMission:  # pylint: disable=too-few-public-methods
```

Comment:
`class MultiCallbackMission(Generic[_T, _R]):`

and `Task` is better than `Mission`